test/test_compile.py

```diff
@@ -0,0 +1,66 @@
+# Copyright Contributors to the Pyro project.
+# SPDX-License-Identifier: Apache-2.0
+
+import pytest
+
+from jax import lax, random
+from jax.lib import xla_bridge
+
+import numpyro
+from numpyro.contrib.autoguide import AutoDiagonalNormal
+import numpyro.distributions as dist
+from numpyro.infer import ELBO, MCMC, NUTS, SVI
+import numpyro.optim as optim
+
+
+GLOBAL = {"count": 0}
+
+
+def model(deterministic=True):
+    GLOBAL["count"] += 1
+    x = numpyro.sample("x", dist.Normal())
+    if deterministic:
+        numpyro.deterministic("x_copy", x)
+
+
+@pytest.mark.parametrize('deterministic', [True, False])
+def test_mcmc_one_chain(deterministic):
+    GLOBAL["count"] = 0
+    mcmc = MCMC(NUTS(model), 100, 100)
+    mcmc.run(random.PRNGKey(0), deterministic=deterministic)
+    mcmc.get_samples()
+
+    if deterministic:
+        assert GLOBAL["count"] == 4
+    else:
+        assert GLOBAL["count"] == 3
```

Comment:
By some small changes, we are able to reduce the number of time we trace the model to 3 :dagger: